rust/lancedb/src/io/object_store.rs

```diff
@@ -62,12 +77,29 @@ impl PrimaryOnly for Path {
     }
 }
 
-/// An object store that mirrors write to secondsry object store first
+/// Controls the behavior of copying objects in the secondary store.
+#[derive(Debug, Clone)]
+pub enum MirroringSecondaryCopy {
+    // Default behaviour is to copy
+    Copy,
+    // Since the secondary store may not be as durable as the primary, the copy source
+    // may exist on the primary but not on the secondary. If the source is not found,
+    // this skips making the copy
+    SkipIfNotFound,
```

Comment:
Generally you'd use `SkipIfNotFound` if you can't be sure that the source object won't be in the secondary store, and your use case allows for the destination not to be there either. As a concrete example, you might want to use `SkipIfNotFound` if you were using the secondary object store as some kind of read through cache. E.g. Maybe your primary object store is s3, but you're mirroring objects to a local ephemeral volume. In this scenario, it could also be that another process is creating objects on the primary store, and if they don't exist on the copy it's not so bad (it's just a cache miss).

Generally you'd use `Copy` if you had some tighter constraints on about maintaining consistency between what's in your primary and secondary stores, and maybe you want the caller to be notified that the copy object couldn't be created on the secondary store. This option being (being the default), is added backwards compatibility.